sentry-android-core/src/main/java/io/sentry/android/core/internal/util/PrivilegeEscalationViaContentProviderChecker.java

```diff
@@ -0,0 +1,63 @@
+package io.sentry.android.core.internal.util;
+
+import android.annotation.TargetApi;
+import android.content.ContentProvider;
+import android.net.Uri;
+import android.os.Build;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+import io.sentry.NoOpLogger;
+import io.sentry.android.core.BuildInfoProvider;
+
+/**
+ * Protects against "Privilege Escalation via Content Provider" (CVE-2018-9492).
+ * <p>
+ * See https://www.cvedetails.com/cve/CVE-2018-9492/
+ * and https://github.com/getsentry/sentry-java/issues/2460
+ */
+@ApiStatus.Internal
+public final class PrivilegeEscalationViaContentProviderChecker {
```

Comment:
I created this SRP class to make it testable. Hopefully, Secure Theorem will recognize that the vulnerability is fixed even though I did not 100% copy-paste their suggested code.

Reply:
Does it make sense to fetch a snapshot into the product and run the analyser against it?

Reply:
@artour-bakiev, I think this is what I'm suggesting in the **Next steps** in the PR description. Or do you mean something else?

Reply:
I thought about running the datatheorem check **before** landing the PR into main branch. But perhaps it was intended, sorry if it was :)